services/wallet/token/balancefetcher/balance_fetcher.go

```diff
@@ -82,7 +82,7 @@ func (bf *DefaultBalanceFetcher) fetchBalancesForChain(parent context.Context, c
 		group.Add(func(parent context.Context) error {
 			balances, err := bf.FetchChainBalances(parent, accounts, ethScanContract, atBlock)
 			if err != nil {
-				return nil
+				return err
```

Comment:
I am not sure what was an idea here and in the lines below, guess @IvanBelyakoff can say more? By my opinion returning err in those cases look ok.